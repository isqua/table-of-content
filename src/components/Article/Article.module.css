.article {
    padding: 48px 32px;
}

.title {
    margin: 0;
    font-size: 42px;
}
