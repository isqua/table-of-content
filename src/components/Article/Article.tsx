import { type PageDescriptor } from '../../features/toc'

import styles from './Article.module.css'

type ArticleProps = {
    page: PageDescriptor
}

export function Article({ page }: ArticleProps) {
    return (
        <article className={styles.article}>
            <h1 className={styles.title}>{page.title}</h1>
        </article>
    )
}
