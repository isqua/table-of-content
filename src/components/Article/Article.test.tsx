import { render, screen } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import { Article } from './Article'

describe('components/Article', () => {
    it('should render an article', () => {
        const page = {
            id: 'foo',
            parentId: '',
            title: 'Foo Article'
        }

        render(<Article breadcrumbs={[page]} page={page} />)

        expect(screen.getByRole('article')).toMatchSnapshot()
    })
})
