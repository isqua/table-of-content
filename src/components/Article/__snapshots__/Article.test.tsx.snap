// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`components/Article > should render an article 1`] = `
<article
  class="_article_a23c90"
>
  <h1
    class="_title_a23c90"
  >
    Foo Article
  </h1>
</article>
`;
